src/overlay/Peer.cpp

```diff
@@ -1163,50 +1224,65 @@ Peer::maybeSendNextBatch()
         }
     }
 
-    CLOG_TRACE(Overlay, "Peer {}: send next flood batch of {}",
-               mApp.getConfig().toShortString(getPeerID()),
-               (oldOutboundCapacity - mOutboundCapacity));
+    CLOG_TRACE(Overlay, "{} Peer {}: send next flood batch of {}",
+               mApp.getConfig().toShortString(
+                   mApp.getConfig().NODE_SEED.getPublicKey()),
+               mApp.getConfig().toShortString(getPeerID()), sent);
 }
 
 void
 Peer::endMessageProcessing(StellarMessage const& msg)
 {
-    if (shouldAbort())
+    if (shouldAbort() || !mFlowControlMessages)
     {
         return;
     }
 
-    mCapacity.mTotalCapacity++;
-    if (mApp.getOverlayManager().isFloodMessage(msg))
+    bool isFloodMessage = mApp.getOverlayManager().isFloodMessage(msg);
+
+    mFlowControlMessages->releaseLocalCapacity(msg);
```

Comment:
maybe `release` should return the amount of capacity released? That way you don't need to have this tight coupling between whatever criteria is used internally (like if it's a flood message) and the call site.

Reply:
Good idea, done.